src/Umbraco.Web.UI.Client/src/views/common/overlays/insert/insert.controller.js

```diff
@@ -0,0 +1,136 @@
+                    vm.macroPickerOverlay = null;
+
+                }
+            };
+
+        }
+
+        function openPageFieldOverlay() {
+            vm.pageFieldOverlay = {
+                title: "Insert value",
+                description: "Select a value from the currentpage",
+                submitButtonLabel: "Insert",
+                closeButtonlabel: "Cancel",
+                view: "insertfield",
+                show: true,
+                submit: function(model) {
+
+                  $scope.model.insert = {
+                      "type": "umbracoField",
+                      "umbracoField": model.umbracoField
+                  };
+
+                  $scope.model.submit($scope.model);
+
+                  vm.pageFieldOverlay.show = false;
+                  vm.pageFieldOverlay = null;
+                },
+                close: function (model) {
+                    vm.pageFieldOverlay.show = false;
+                    vm.pageFieldOverlay = null;
+                }
+            };
+        }
+
+        function openDictionaryItemOverlay() {
+
+            vm.dictionaryItemOverlay = {
+                view: "treepicker",
+                section: "settings",
+                treeAlias: "dictionary",
+                entityType: "dictionary",
+                multiPicker: false,
+                title: "Insert dictionary item",
+                show: true,
+                select: function(node){
+
+                    $scope.model.insert = {
+                        "type": "dictionary",
+                        "node": node
+                    };
+
+                    $scope.model.submit($scope.model);
+
+                    vm.dictionaryItemOverlay.show = false;
+                    vm.dictionaryItemOverlay = null;
+                },
+
+                close: function(model) {
+                    vm.dictionaryItemOverlay.show = false;
+                    vm.dictionaryItemOverlay = null;
+                }
+            };
+        }
+
+        function openPartialOverlay() {
+            vm.partialItemOverlay = {
+                view: "treepicker",
+                section: "settings", 
+                treeAlias: "partialViews",
+                entityType: "partialView",
+                multiPicker: false,
+                show: true,
+                title: "Insert partial view",
+
+                select: function(node){
+                    
+                    $scope.model.insert = {
+                        "type": "partial",
+                        "node": node
+                    };
+
+                    $scope.model.submit($scope.model);
+
+                    vm.partialItemOverlay.show = false;
+                    vm.partialItemOverlay = null;
+                },
+
+                close: function (model) {
+                    vm.partialItemOverlay.show = false;
+                    vm.partialItemOverlay = null;
+                }
+            };
+        }
+
+    }
+
```

Comment:
Code all looks fine in this controller for the insert overlay - but anyone who has lang of their user set to Danish gonna get English. So lets get them from the language please